### PR TITLE
Refactor record trimming code (filter-abund*, trim-low-abund) to use library functions.

### PR DESCRIPTION
**Pull request by ctb**
Refactor for extraction of common record trimming code, eliminate ThreadedSequenceProcessor, and use broken_paired_reader for record loading.

Should be merged after #1372, because of fixes for PEP8 errors in khmer/utils.py.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the `ChangeLog`?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)
- [x] Is the Copyright year up to date?

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review @betatim @standage @luizirber.

**Comment by ctb** [comment]
OK, now really ready for review @betatim @standage @luizirber :)

File: khmer/utils.py [hunk 1]
```diff
@@ -252,21 +252,21 @@ def write_record_pair(read1, read2, fileobj):
 class ReadBundle(object):
     def __init__(self, *raw_records):
         self.reads = [i for i in raw_records if i]
-        self.cleaned_reads = [r.sequence.replace('N', 'A') for r in self.reads]
+        self.cleaned_seqs = [r.sequence.replace('N', 'A') for r in self.reads]
```
Comment (standage) [inline khmer/utils.py]: 👍

**Comment by ctb** [comment]
Excellent, thanks!

**Pull request merged**